packages/dds/merge-tree/docs/high-level.md

```diff
@@ -0,0 +1,401 @@
+
+1. `insert(index: number, text: string)`
+2. `removeRange(start: number, end: number)`
+3. `annotateRange(start: number, end: number, properties: object)`
+
+You can insert text, remove a range of text, or give a key-value property to a range of text. 
+
+The API of the merge-tree is based around these operations. These can be thought of as JSON payloads containing the above function parameters. For example, `{ type: "insert", index: 0, text: "foo" }`.
+
+In addition to these parameters, each operation ("**op**") has a **sequence number** associated with it, abbreviated "seq". This is a [strictly increasing] integer that (simplifying a bit)[^1] is unique to a given operation. The sequence number determines the order in which operations get applied. That is, the operation at seq n _must_ be processed before the op at seq n+1. 
+
+If you are familiar with [sequence numbers in the context of network protocols like TCP], this is very similar.
+
+The exact machinery for sequencing ops lives outside of merge-tree and is not necessary for understanding the core algorithms. 
+
+If we look at a simplified example of an op stream:
+
+```json
+[
+    { "type": "insert", "index": 0, "text": "hello", "seq": 1 },
+    { "type": "remove", "start": 1, "end": 5, "seq": 2 },
+    { "type": "insert", "index": 1, "text": "i", "seq": 3 },
+]
+```
+
+Here, we have a sequence of ops that constructs the following string:
+
+seq 1: "hello" (insert string "hello")  
+seq 2: "h" (remove range `[1-5)`)  
+seq 3: "hi" (insert string "i")  
+
+This is relatively straightforward behavior to reason about, and should be roughly what you would expect to see from a regular string data structure used for normal text editing.
+
+The complexity of merge-tree comes from its support for collaboration. In order to talk about collaboration, we have to add 2 more fields to our ops: **client id** (`clientId`) and **reference sequence number** (`refSeq`).
+
+The clientId is exactly what it sounds like: a unique identifier for a particular user in a collaboration session. If three people are editing a document in their browser at the same time, each user would be given a unique clientId. In practice these are UUIDs, though merge-tree has some optimizations that convert the UUIDs to small integers by [interning] them.
+
+The reference sequence number is slightly more complex, and will require a bit more background to fully describe.
+
+In general, collaboration looks something like this:
+
+One user, or client, will create an op and send it to the server. The server will process the op and assign it a sequence number. For the client that sent the op, the server will acknowledge the op and send that client back a sequence number. For the other clients, it will send them this op with the sequence number attached.
+
+All clients then process this op and update their refSeq. The refSeq is the sequence number of the last operation that a particular client processed from the server.
```

Comment:
I think I would explain this section a bit differently. The explanation here is somewhat at risk for someone new to distributed systems being under the impression that "all clients" might perform the action here simultaneously in some sense, which is not actually true.

I think you can lean on the fact that the reference sequence number is determined locally (i.e. by the client which submits the op) to explain it earlier. Alternatively, if wanting to explain collaboration at the level of detail required here (which I think could be justifiable), in lieu of this paragraph I might go with something more like:

> Upon receipt of this op, other clients (who may have their own unacked operations) must reason about how to apply the op. To do so, they must have information about the context in which the op was applied. `referenceSequenceNumber` is exactly this context: all clients keep track of the most recent sequence number operation from the server that they have processed. Whenever a client submits an op, it includes that most recently processed sequence number as part of the op. This is known as the op's `referenceSequenceNumber`.

Also, if you're looking to point toward external references in this doc for more information, the academic term in a general sense is 'logical clock'.